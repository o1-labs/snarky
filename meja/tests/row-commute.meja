let f = fun (x1 : [> `A],
             x2 : [> `B],
             x3 : [> `C],
             x4 : [> `D],
             x5 : [< `A | `B | `C],
             b) => {
  let y1 = if (b) { x1; } else { x2; };
  let y2 = if (b) { x3; } else { x4; };
  let y3 = if (b) { x2; } else { x4; };
  let y4 = if (b) { x1; } else { x5; };
  (y1, y2, y3, y4);
};
