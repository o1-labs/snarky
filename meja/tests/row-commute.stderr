File "tests/row-commute.meja", line 10, characters 33-35:
Error: Incompatible types ([>`A | `B | `C | `D] as 'a) and
       ([<`A | `B | `C] as 'b):
       Cannot unify ([>`A | `B | `C | `D] as 'a) and ([<`A | `B | `C] as 'b):
       the constructor `D is not present in both.
